### ISSUE TITLE
Package conversion crashed at solution refresh, package refs removed/not added.

### ISSUE THREAD

**Issue by OsirisTerje** [twsouthwick/NuGetPackageConfigConverter#25]
### Installed product versions
- Visual Studio: VS 2017, 15.5.2
- C# class libs, test projects (most are std C# class libs, a few have the unti test id in there,), some vcxproj managed C++.
- Some projects have the new csproj format, some have the old with package.config.  

### Description
Converting a solution with 74 projects.  All projects were converted, but it seemed to crash at the end, when it was reloading the solution.  Visual Studio survived, so only the extension crashed (that was very good).  When I reloaded the solution, the packages.config files were gone, and 
1) The csproj files that were in the new format , had their section with package references removed.  
![image](https://user-images.githubusercontent.com/203432/34942669-c12c1134-f9f8-11e7-9349-1a0b19a382d3.png)

2. For the projects with the old format, the references were removed (as they should), but the new packagereferences were not added. 

### Current behavior
Converted a few 
![image](https://user-images.githubusercontent.com/203432/34942252-39fbdd3a-f9f7-11e7-9405-049765869d04.png)

Then:
![image](https://user-images.githubusercontent.com/203432/34942258-3ebc6cea-f9f7-11e7-979b-bc33cedc4731.png)

### Expected behavior
Convert them all :D

From Clipboard:
System.AggregateException: One or more errors occurred. ---> System.NullReferenceException: Object reference not set to an instance of an object.
   at NuGetPackageConfigConverter.PackageManagerConverter.AddRestoreProjectStyle(String path) in C:\projects\nugetpackageconfigconverter\src\NuGetPackageConfigConverter\Services\PackageManagerConverter.cs:line 204
   at NuGetPackageConfigConverter.PackageManagerConverter.RefreshSolution(Solution sln, IEnumerable`1 projects) in C:\projects\nugetpackageconfigconverter\src\NuGetPackageConfigConverter\Services\PackageManagerConverter.cs:line 193
   at NuGetPackageConfigConverter.PackageManagerConverter.<>c__DisplayClass7_0.<ConvertAsync>b__0(ConverterUpdateViewModel model, CancellationToken token) in C:\projects\nugetpackageconfigconverter\src\NuGetPackageConfigConverter\Services\PackageManagerConverter.cs:line 63
   at NuGetPackageConfigConverter.DialogWindowConverterViewProvider.<>c__DisplayClass0_2.<ShowAsync>b__0() in C:\projects\nugetpackageconfigconverter\src\NuGetPackageConfigConverter\Services\DialogWindowConverterViewProvider.cs:line 37
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.NullReferenceException: Object reference not set to an instance of an object.
   at NuGetPackageConfigConverter.PackageManagerConverter.AddRestoreProjectStyle(String path) in C:\projects\nugetpackageconfigconverter\src\NuGetPackageConfigConverter\Services\PackageManagerConverter.cs:line 204
   at NuGetPackageConfigConverter.PackageManagerConverter.RefreshSolution(Solution sln, IEnumerable`1 projects) in C:\projects\nugetpackageconfigconverter\src\NuGetPackageConfigConverter\Services\PackageManagerConverter.cs:line 193
   at NuGetPackageConfigConverter.PackageManagerConverter.<>c__DisplayClass7_0.<ConvertAsync>b__0(ConverterUpdateViewModel model, CancellationToken token) in C:\projects\nugetpackageconfigconverter\src\NuGetPackageConfigConverter\Services\PackageManagerConverter.cs:line 63
   at NuGetPackageConfigConverter.DialogWindowConverterViewProvider.<>c__DisplayClass0_2.<ShowAsync>b__0() in C:\projects\nugetpackageconfigconverter\src\NuGetPackageConfigConverter\Services\DialogWindowConverterViewProvider.cs:line 37<---

**Comment by twsouthwick** [comment]
Can't say for certain without a repro, but I'm betting it's similar to the fix put in with https://github.com/twsouthwick/NuGetPackageConfigConverter/commit/f76c7614f56dae8d752327f316679068f1b14ebd. Some project types do not implement `Project.FullName`. I created a method to catch the method, but didn't realize I was using it elsewhere. The call to `.FullName` here:

https://github.com/twsouthwick/NuGetPackageConfigConverter/blob/89e7d9eb77fdc09cd2e8394349a0446186dc32c1/src/NuGetPackageConfigConverter/Services/PackageManagerConverter.cs#L188

Should be replaced with the call defined here (with making it a public extension method instead of a private helper):

https://github.com/twsouthwick/NuGetPackageConfigConverter/blob/89e7d9eb77fdc09cd2e8394349a0446186dc32c1/src/NuGetPackageConfigConverter/VSExtensions.cs#L32-L42

Any other instance of this should be found as well.

**Comment by twsouthwick** [comment]
Also will need to filter for `null` values.

@OsirisTerje are you interested in submitting a PR? I see you forked the repo

**Comment by OsirisTerje** [comment]
Hi!  I'm checking it out now, and have also fixed a few other issues I found, and also added some more output to make it easier to see the progress.  
And yes, I'll submit a PR when it works :-)

**Comment by OsirisTerje** [comment]
@twsouthwick :  I got it working now, and have tried it on 3 different bigger mixed solutions, and it is working.  I have seen exceptions installing packages into one project, but it worked passed those.  I could not see what was different there, so just did it manually in that case.  
I am a bit unsure about the sln reload, if there is ample time to really reload all projects. I added a delay there, but would be better if there was some way to really ensure the solution was fully loaded. 

The log files are under %temp%/NuGetPackageConfigConverter folder.  They are flushed as it writes, so if it hangs, you will see where it stops. 
The updates to the UI will also help wrt that. 

This fix may also fix #16 , at least make it easier to see where it stops and why.
